drivers/ethernet/eth_stm32_hal.c

```diff
@@ -26,10 +26,27 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #include "eth_stm32_hal_priv.h"
 
+#if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
+    !defined(DT_DTCM_BASE_ADDRESS)
+#error DTCM for DMA buffer is activated but DT_DTCM_BASE_ADDRESS is not present
+#endif
+
+#if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
+    defined(DT_DTCM_BASE_ADDRESS)
+static ETH_DMADescTypeDef
+	dma_rx_desc_tab[ETH_RXBUFNB] __aligned(4) __dtcm_noinit_section;
```

Comment:
Is `__aligned(4)` required when you're defining section with `SUBALIGN(4)` in linker?

Reply:
Good point. I don't think so. DMA with nocache would need a 32 alignment but I couldn't find any restrictions for DMA and DTCM.